task2c

zdelena opened the pull request:
Задание 1: Создайте функцию maxOfTwo, которая принимает два числа в качестве аргументов и возвращает наибольшее число. Затем, используйте эту функцию для поиска максимального числа из двух разных пар чисел.

Задание 2: Создайте функцию sayHello, которая принимает строковый аргумент name и возвращает приветственное сообщение. Функция должна иметь параметр по умолчанию "мир". Затем вызовите функцию дважды: с именем и без имени.

Задание 3: Создайте инфиксную функцию isGreater , которая принимает два числа и возвращает true, если первое число больше второго, и false в противном случае. Затем используйте эту функцию, сравнив две пары чисел.

Задание 4: Создайте рекурсивную функцию generateSequence, которая принимает целое число n и генерирует последовательность чисел от 1 до n. Используйте аннотацию tailrec для оптимизации.

Задание 5: Создайте список имен и используйте функции let, run, also, apply и with для выполнения различных операций над этим списком (например, добавьте имя, удалите имя, переверните список и т. д.).

![изображение](https://github.com/user-attachments/assets/636961fb-34ea-42fb-b2f1-d732d44ffef8)

zdelena committed:
task2c

zdelena committed:
fixed task 2c

semenihina-surf commented on app/src/main/kotlin/task2c.kt:
```diff
@@ -0,0 +1,126 @@
+     */
+    fun maxOfTwo(a: Int, b: Int): Int {
+        return if (a > b) {
+            a
```
можно в одну строку return if (a > b) a else b

semenihina-surf commented on app/src/main/kotlin/task2c.kt:
```diff
@@ -0,0 +1,126 @@
+    // и генерирует последовательность чисел от 1 до n.
+    // Используйте аннотацию tailrec для оптимизации.
+
+    tailrec fun genSequence(n: Int, i: Int = 1): Int {
```
по заданию "generateSequence"

semenihina-surf replied:
можно ввести строковую переменную, например, acc: String = "" 
куда будет сохраняться последовательность и использовать trim 

как будет выглядеть, учитывая комментарии ?

zdelena committed:
alternative recursion

zdelena committed:
fix after review

поправила return в 1 строку, добавила альтернативное решение для рекурсии (со строкой)
